src/objective-c/GRPCClient/private/GRPCCompletionQueue.m

```diff
@@ -61,22 +70,24 @@ - (instancetype)init {
       gDefaultConcurrentQueue = dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0);
     });
     dispatch_async(gDefaultConcurrentQueue, ^{
+      gpr_timespec deadline = gpr_time_from_seconds(lTimeoutSecs, GPR_CLOCK_REALTIME);
       while (YES) {
-        // The following call blocks until an event is available.
-        grpc_event event = grpc_completion_queue_next(unmanagedQueue,
-                                                      gpr_inf_future(GPR_CLOCK_REALTIME),
-                                                      NULL);
+        // The following call blocks until an event is available or the deadline elapses.
+        grpc_event event = grpc_completion_queue_next(unmanagedQueue, deadline, NULL);
         GRPCQueueCompletionHandler handler;
         switch (event.type) {
-          case GRPC_OP_COMPLETE:
+          case GRPC_OP_COMPLETE: // Falling through deliberately
+          case GRPC_QUEUE_TIMEOUT:
             handler = (__bridge_transfer GRPCQueueCompletionHandler)event.tag;
-            handler(event.success);
+            if (handler) {
```

Comment:
What's the tag in case of timeout? And if it's `NULL`, why bother falling through and adding the `if` block?